### DISCUSSION TITLE
A Quick First Look at Amazon Bedrock (with Node.js)

### DISCUSSION THREAD

**Discussion by giscus** [cfjedimaster/raymondcamden2023#38] [Comments]
# A Quick First Look at Amazon Bedrock (with Node.js)

A look at Amazon Bedrock, Amazon's generative AI service.

https://www.raymondcamden.com/2024/04/04/a-quick-first-look-at-amazon-bedrock-with-nodejs

**Comment by RoyDumblauskas** [comment]
If you want the response body to be text, use `ConverseCommand`, not the `InvokeModelCommand`.

**Reply by cfjedimaster** [reply]
Oh nice, thank you!

**Comment by charlesr1971** [comment]
Great article. I am currently working with MCP? 
Do you know anything about how I can send MCP tools to Bedrock, rather just a prompt? 

So, tools work in conjunction with prompts. 

Using OpenAi, we do something like:

```
const response = await this.openai.responses.create({
			model: this.model,
			input: messages,
			tools: this.tools,
});
```

This sends a message and a selection of tools to OpenAi. 
The AI then decides which tool to use and then sends back a response, based on the message sent and the tool chosen. 

So a real world example might be that I want to summarise a PDF. 
After the PDF is uploaded, a text file is generated from the file using Apache Tika, and then this text is sent to Bedrock along with some tools. One of the tools asks the AI to summarise the text and return the output. The AI will understand from the context that it should use the summarisation tool etc

**Reply by cfjedimaster** [reply]
I'm sorry but I'm useless here. :) I mainly focus on Gemini and OpenAI these days, I've yet to play more with Bedrock.
